server/models/Transaction.js

```diff
@@ -257,6 +258,16 @@ function defineModel() {
       },
 
       hooks: {
+        beforeCreate: async transaction => {
+          if (!transaction.description) {
+            transaction.description = await generateDescription(transaction);
```

Comment:
This centralized approach will make it more difficult to wrap the creation of new transactions in SQL transactions (what was started in https://github.com/opencollective/opencollective-api/pull/5981). We can eventually take the one from `beforeCreate` arguments and pass it down to `generateDescription` but that will make things more complex (what if you're using loaders?).

In this regard, the existing approach of setting the description where the transaction gets created makes more sense as these parts of the code already have the data needed to generate the descriptions, they don't need to preload any additional information.

We can still add some helpers when they're used in multiple places, for example:

```
models.Transaction.create({
  isRefund: true,
  description: generateRefundTransactionDescription(originalTransaction),
})
```

Reply:
Ok, let's take some time to think about this.